swift/Samples/ModalContainer/Sources/ModalContainerScreen.swift

```diff
@@ -0,0 +1,51 @@
+import WorkflowUI
+
+/// A `ModalContainerScreen` displays a base screen and optionally one or more modals on top of it.
+public struct ModalContainerScreen<BaseScreen: Screen>: Screen
+{
+
+    /// The base screen to show underneath any modally presented screens.
+    public let baseScreen: BaseScreen
+
+    /// Modally presented screens
+    public let modals: [ModalContainerScreenModal]
+
+    public init(baseScreen: BaseScreen, modals: [ModalContainerScreenModal]) {
+        self.baseScreen = baseScreen
+        self.modals = modals
+    }
+
+    public func viewControllerDescription(environment: ViewEnvironment) -> ViewControllerDescription {
+        return ModalContainerViewController.description(for: self, environment: environment)
+    }
+}
+
+/// Represents a single screen to be displayed modally
+public struct ModalContainerScreenModal {
```

Comment:
Can we call this simply `Modal`? The fully-qualified type will include the `ModalContainer`.

Reply:
Will keep a note of that for any subsequent cleanup. Thanks!